### PR TITLE
cukes and minor bugs

### PR DESCRIPTION
**Pull request by xithan**
- suppress useless warning
- minor bugfixes
- add csv export view "nested_fields"
- speed up cucumber features

### CODE REVIEW

File: card/mod/standard/set/abstract/search.rb [hunk 1]
```diff
@@ -92,14 +92,58 @@ def implicit_item_view
...
+    columns = []
+    Card.fetch(list.first).format.each_nested_field do |chunk|
+      columns << chunk.referee_name.tag
+    end
+    list.map do |item_name|
+      row = [item_name]
+      row +=
+        columns.map do |field|
+          nest [item_name, field], view: :core
+        end
```
Comment (ethn) [inline card/mod/standard/set/abstract/search.rb]: would `each_with_object` make this cleaner?

File: card/mod/standard/set/abstract/search.rb [hunk 2]
```diff
@@ -92,14 +92,58 @@ def implicit_item_view
...
+  def column_titles extra_titles=nil
+    res = Array extra_titles
+    card1 = search_with_params(limit: 1).first
+    card1.each_nested_chunk do |chunk|
+      opts = chunk.options
+      res << if %w(name link).member? opts[:view]
+               opts[:view]
+             else
+               opts[:nest_name].to_name.tag
+             end
```
Comment (ethn) [inline card/mod/standard/set/abstract/search.rb]: another method?  `def column_title view, nest_name`?

File: card/config/initializers/core_extensions.rb [hunk 3]
```diff
@@ -2,13 +2,14 @@
 
 # extend core Ruby object classes
 
-class Class
+class Module
```
Comment (xithan) [inline card/config/initializers/core_extensions.rb]: it does
`Class.ancestors => [Class, Module, Object, Kernel, BasicObject]`

File: card/mod/carrierwave/set/abstract/attachment/paths.rb [hunk 4]
```diff
@@ -22,7 +22,9 @@ def coded_dir new_mod=nil
 
 def mod_dir new_mod=nil
   find_mod = new_mod || mod
+
   Card::Mod::Loader.mod_dirs.path(find_mod) ||
+    (find_mod.to_sym == :test && "test") ||
     raise(Error, "can't find mod \"#{find_mod}\"")
```
Comment (xithan) [inline card/mod/carrierwave/set/abstract/attachment/paths.rb]: that's a different logic but I can change it to
```
mod_name = new_mod || mod
dir = Card::Mod::Loader.mod_dirs.path(mod_name) || (mod_name == :test && "test")
raise Error, "can't find mod \"#{find_mod}\"" unless dir
dir
```

File: card/mod/standard/set/abstract/search.rb [hunk 5]
```diff
@@ -92,14 +92,58 @@ def implicit_item_view
 end
 
 format :csv do
-  view :card_list do |args|
-    items = super args
+  view :nested_fields do
```
Comment (xithan) [inline card/mod/standard/set/abstract/search.rb]: @ethn Any thoughts on the view name? Maybe just "fields" or "export_items", "item_fields"?

**Comment by xithan** [comment]
Re review status: the official description is 
**Comment**: Submit general feedback without explicitly approving the changes or requesting additional changes.
**Approve**: Submit feedback and approve merging the changes proposed in the pull request.
**Request changes**: Submit feedback that must be addressed before the pull request can be merged.
I would say we never merge a PR with "requested changes". You can change the status of your review any time later to "approved" (to see it positive: that means you care for the changes you suggested and you acknowledge the efforts that has been made to address your comments). If you think there is no need for a further discussion and your suggested changes are optional or if we loose another day until you can approve the changes because of the time shift and you think think it's not worth it then choose approve. I will check your comments anyway no matter what status you choose :smile:

**Pull request merged**